Source/SwiftLintBuiltInRules/Rules/Lint/MarkRule.swift

```diff
@@ -9,239 +10,143 @@ struct MarkRule: CorrectableRule {
-        ]
+        nonTriggeringExamples: MarkRuleExamples.nonTriggeringExamples,
+        triggeringExamples: MarkRuleExamples.triggeringExamples,
+        corrections: MarkRuleExamples.corrections
     )
+}
 
-    private let spaceStartPattern = "(?:\(nonSpaceOrTwoOrMoreSpace)\(mark))"
-
-    private let endNonSpacePattern = "(?:\(mark)\(nonSpace))"
-    private let endTwoOrMoreSpacePattern = "(?:\(mark)\(twoOrMoreSpace))"
-
-    private let invalidEndSpacesPattern = "(?:\(mark)\(nonSpaceOrTwoOrMoreSpace))"
-
-    private let twoOrMoreSpacesAfterHyphenPattern = "(?:\(mark) -\(twoOrMoreSpace))"
-    private let nonSpaceOrNewlineAfterHyphenPattern = "(?:\(mark) -[^ \n])"
-
-    private let invalidSpacesAfterHyphenPattern = "(?:\(mark) -\(nonSpaceOrTwoOrMoreSpaceOrNewline))"
+private extension MarkRule {
+    final class Visitor: ViolationsSyntaxVisitor<ConfigurationType> {
+        override func visit(_ token: TokenSyntax) -> SyntaxVisitorContinueKind {
+            for result in token.violationResults() {
+                violations.append(result.position)
+            }
+            return .skipChildren
+        }
+    }
 
-    private let invalidLowercasePattern = "(?:// ?[Mm]ark:)"
+    final class Rewriter: ViolationsSyntaxRewriter {
+        private var correctedPositions: Set<AbsolutePosition> = []
 
-    private let missingColonPattern = "(?:// ?MARK[^:])"
-    // The below patterns more specifically describe some of the above pattern's failure cases for correction.
-    private let oneOrMoreSpacesBeforeColonPattern = "(?:// ?MARK +:)"
-    private let nonWhitespaceBeforeColonPattern = "(?:// ?MARK\\S+:)"
-    private let nonWhitespaceNorColonBeforeSpacesPattern = "(?:// ?MARK[^\\s:]* +)"
-    private let threeSlashesInsteadOfTwo = "/// MARK:?"
+        override func visit(_ token: TokenSyntax) -> TokenSyntax {
+            var newToken = token
 
-    private var pattern: String {
-        return [
-            spaceStartPattern,
-            invalidEndSpacesPattern,
-            invalidSpacesAfterHyphenPattern,
-            invalidLowercasePattern,
-            missingColonPattern,
-            threeSlashesInsteadOfTwo
-        ].joined(separator: "|")
-    }
+            var pieces = newToken.leadingTrivia.pieces
+            for result in newToken.violationResults(withCorrect: true) {
+                correctionPositions.append(result.position)
+                result.correct?(&pieces)
+            }
 
-    func validate(file: SwiftLintFile) -> [StyleViolation] {
-        return violationRanges(in: file, matching: pattern).map {
-            StyleViolation(ruleDescription: Self.description,
-                           severity: configuration.severity,
-                           location: Location(file: file, characterOffset: $0.location))
+            if pieces != newToken.leadingTrivia.pieces {
+                newToken.leadingTrivia = .init(pieces: pieces)
+            }
+            return super.visit(newToken)
```

Comment:
```suggestion
            return super.visit(token.with(\.leadingTrivia, Trivia(pieces: pieces)))
```

That way, you can omit the explicit copy `newToken` and the previous `if` statement.